rust/lance/src/dataset/fragment.rs

```diff
@@ -504,10 +504,13 @@ impl FileFragment {
 
         // scan with predicate and row ids
         let mut scanner = self.scan();
-        scanner
-            .with_row_id()
-            .filter(predicate)?
-            .project::<&str>(&[])?;
+        scanner.with_row_id();
+
+        // if predicate is true, delete the whole fragment
+        // else, filter the predicate
+        if predicate != "true" {
+            scanner.filter(predicate)?.project::<&str>(&[])?;
+        }
```

Comment:
It's perhaps silly, but it might also be good to support `false`:

```suggestion
        } else if &predicate.trim().to_lowercase() == "false" {
            return Ok(Some(self));
        }
```